platform-sdk/swirlds-unit-tests/common/swirlds-common-test/src/test/java/com/swirlds/common/test/metrics/extensions/PhaseTimerTests.java

```diff
@@ -0,0 +1,592 @@
+    }
+
+    @Test
+    @DisplayName("Only Absolute Time Metrics Enabled Test")
+    void onlyAbsoluteTimeMetricsTest() {
+        final AtomicBoolean fooMetricRegistered = new AtomicBoolean(false);
+        final List<Double> fooUpdates = new ArrayList<>();
+
+        final AtomicBoolean barMetricRegistered = new AtomicBoolean(false);
+        final List<Double> barUpdates = new ArrayList<>();
+
+        final AtomicBoolean bazMetricRegistered = new AtomicBoolean(false);
+        final List<Double> bazUpdates = new ArrayList<>();
+
+        final Metrics metrics = buildMockMetrics(config -> {
+            assertEquals("test", config.getCategory());
+            assertEquals(RunningAverageMetric.Config.class, config.getClass());
+            if (config.getName().equals("TestPhases_time_FOO")) {
+                assertFalse(fooMetricRegistered.get());
+                fooMetricRegistered.set(true);
+                return buildMockRunningAverageMetric(fooUpdates);
+            } else if (config.getName().equals("TestPhases_time_BAR")) {
+                assertFalse(barMetricRegistered.get());
+                barMetricRegistered.set(true);
+                return buildMockRunningAverageMetric(barUpdates);
+            } else if (config.getName().equals("TestPhases_time_BAZ")) {
+                assertFalse(bazMetricRegistered.get());
+                bazMetricRegistered.set(true);
+                return buildMockRunningAverageMetric(bazUpdates);
+            } else {
+                throw new AssertionError("Unexpected metric name: " + config.getName());
+            }
+        });
+
+        final PlatformContext platformContext =
+                TestPlatformContextBuilder.create().withMetrics(metrics).build();
+        final FakeTime time = new FakeTime();
+
+        final PhaseTimer<TestPhases> timer = new PhaseTimerBuilder<>(platformContext, time, "test", TestPhases.class)
+                .enableAbsoluteTimeMetrics()
+                .setAbsoluteTimeUnit(TimeUnit.UNIT_SECONDS)
+                .build();
+
+        assertTrue(fooMetricRegistered.get());
+        assertTrue(barMetricRegistered.get());
+        assertTrue(bazMetricRegistered.get());
+
+        time.tick(Duration.ofSeconds(1));
+        timer.activatePhase(FOO);
+        time.tick(Duration.ofSeconds(2));
+        timer.activatePhase(BAR);
+        time.tick(Duration.ofSeconds(3));
+        timer.activatePhase(BAZ);
+        time.tick(Duration.ofSeconds(4));
+        timer.activatePhase(BAZ);
+        time.tick(Duration.ofSeconds(5));
+        timer.activatePhase(BAR);
+        time.tick(Duration.ofSeconds(6));
+        timer.activatePhase(FOO);
+
+        for (int i = 0; i < 100; i++) {
+            time.tick(Duration.ofSeconds(1));
+            timer.activatePhase(BAR);
+            time.tick(Duration.ofSeconds(1));
+            timer.activatePhase(BAZ);
+        }
+
+        assertEquals(2, fooUpdates.size());
+        assertEquals(fooUpdates.get(0), 3.0);
```

Comment:
nit: you have `actual, expected` in many places, where it ought to be `expected, actual`

Reply:
fixed